### ISSUE TITLE
Issue inside 'as_dict()' in transactions.py

### ISSUE THREAD

**Issue by hiddenvs** [1200wd/bitcoinlib#329]
I am trying to use bitcoinlib to parse transactions and found problem inside transactions.py file
in function as_dict(), where by return of public_key is this error:

/lib/python3.10/site-packages/bitcoinlib/transactions.py", line 1346, in as_dict
    'public_key': self.public_key.hex(),

AttributeError: 'str' object has no attribute 'hex'

I think it should be without .hex(), but I am not 100% sure about.

**Comment by mccwdev** [comment]
The problem is not with the hex() method, the self.public_key should be always in bytes format.
Could you maybe post the code which leads to this error?

**Comment by Lucshine** [comment]
> The problem is not with the hex() method, the self.public_key should be always in bytes format. Could you maybe post the code which leads to this error?

When the input of the transaction is P2PK, [transactions.py#L1193](https://github.com/1200wd/bitcoinlib/blob/master/bitcoinlib/transactions.py#L1193) will be entered and the public_hex in the script is assigned to self.public_key. The value type here is str type. This problem can be solved by adding a to_bytes function on the right side of the equation.
The code to reproduce this error is as follows:

```python
from bitcoinlib.transactions import Transaction

t = Transaction.parse_hex('01000000010000000000000000000000000000000000000000000000000000000000000000ffffffff07044c86041b0146ffffffff0100f2052a01000000434104e18f7afbe4721580e81e8414fc8c24d7cfacf254bb5c7b949450c3e997c2dc1242487a8169507b631eb3771f2b425483fb13102c4eb5d858eef260fe70fbfae0ac00000000')
t.as_dict()
```

**Comment by mccwdev** [comment]
Thanks, fixed in commit ae354c9
